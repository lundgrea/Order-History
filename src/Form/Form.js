import React, { Component } from 'react';
import './Form.css';


class Form extends Component {
  constructor() {
    super();
    this.state = {
      img: '',
      name: '',
      description: '',
      price: ''
    }
  }

  handleFormInput = event => {
    this.setState({[event.target.name]: event.target.value})
  }

  submitOrder = event => {
    event.preventDefault();
    const newOrder = {
      id: Date.now(),
      ...this.state
    }
    this.props.addOrder(newOrder);
    this.clearAllInputFields()
  }

  clearAllInputFields = () => {
    this.setState(
      { 
      img: '',
      name: '',
      description: '',
      price: ''
    })
  }

  render() {
    return( 
      <form>
        <input
        type="text"
        placeholder="Image Url"
        name="img"
        value={this.state.img}
        onChange={event => this.handleFormInput(event)}
        />
        <input
         type="text"
         placeholder="Name:"
         name="name"
         value={this.state.name}
         onChange={event => this.handleFormInput(event)}
         />
        <input
         type="text"
         placeholder="Description:"
         name="description"
         value={this.state.description}
         onChange={event => this.handleFormInput(event)}/>
        <input
         type="text"
         placeholder="Price:"
         name="price"
         value={this.state.price}
         onChange={event => this.handleFormInput(event)}/>
        <button onClick={event=> this.submitOrder(event)}>Add Purchase</button>
      </form>
    )
    }
  }


export default Form