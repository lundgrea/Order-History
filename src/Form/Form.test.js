import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Form addOrder={jest.fn()}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state when handleFormInput is called', () => {
    const mockEvent = {target: {name: 'img', value: 'the img URL input value'}}
    const expectedURL = 'the img URL input value';
    wrapper.instance().handleFormInput(mockEvent);
    expect(wrapper.state('img')).toEqual(expectedURL)
  })

})