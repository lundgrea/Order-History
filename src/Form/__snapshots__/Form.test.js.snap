// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<form>
  <input
    name="img"
    onChange={[Function]}
    placeholder="Image Url"
    type="text"
    value=""
  />
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name:"
    type="text"
    value=""
  />
  <input
    name="description"
    onChange={[Function]}
    placeholder="Description:"
    type="text"
    value=""
  />
  <input
    name="price"
    onChange={[Function]}
    placeholder="Price:"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Purchase
  </button>
</form>
`;
